### ISSUE TITLE
Healthkit permissions prompt closes instantly after opening (iOS version 17.1.1)

### ISSUE THREAD

**Issue by Mayank-Yadav89** [agencyenterprise/react-native-health#349]
**Describe the bug**
Apple Healthkit permission prompt in iOS version >17 the modal closes instantly after opening up without user interaction.

**To Reproduce**
For iOS version 17 or higher, try running **AppleHealthKit.initHealthKit** function

1. Call AppleHealthKit.initHealthKit
2. Wait for modal to appear
3. Modal closes instantly
4. Put the app to background
5. Open app again: Permission popup comes back and starts working normally.

**Expected behavior**
The modal should not close. The user should be able to enable/disable as usual.

**Screenshots**

**Smartphone:**
- Device: iPhone
- OS: iOS 17.1.1
- Version: "react-native-health": "^1.14.0"

**Additional context**
Works fine on iPhone 7 Plus v15.8

**Comment by MUZAFFAR-PURELOGICS** [comment]
Same issue i am also facing on IOS 17.6.1

**Comment by Iamshivam-dev** [comment]
I am facing the same issue, @MUZAFFAR-PURELOGICS  did you found any solution?

**Comment by MUZAFFAR-PURELOGICS** [comment]
@Iamshivam-dev,
Nope, i did not get any solution, still waiting for administrator to resolve this issue.
